projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/agents-button/agents-button.component.spec.ts

```diff
@@ -61,26 +78,9 @@ describe('AgentsButtonComponent', () => {
     fixture = TestBed.createComponent(AgentsButtonComponent);
     component = fixture.componentInstance;
     store = TestBed.inject(MockStore);
-    agents$ = new Subject<AgentPaging>();
+    agents$ = new Subject<AgentWithAvatar[]>();
     spyOn(TestBed.inject(AgentService), 'getAgents').and.returnValue(agents$);
-    agentPaging = {
-      list: {
-        entries: [
-          {
-            entry: {
-              id: '1',
-              name: 'HR Agent'
-            }
-          },
-          {
-            entry: {
-              id: '2',
-              name: 'Policy Agent'
-            }
-          }
-        ]
-      }
-    };
+    agentWithAvatar = agentWithAvatarList;
```

Comment:
do we need both - agentWithAvatar and agentWithAvatarList? I guess you did that to "reset" this field and that's correct but then it's still not reset because in that way you assign reference too so both variables has the same reference I think. 

Instead of these both variables you can assign this array to agentWithAvatar directly in beforeEach instead of through variable